zeppelin-test/src/main/java/org/apache/zeppelin/test/DownloadUtils.java

```diff
@@ -0,0 +1,537 @@
+            .setInitialMax(completeFileSize)
+            .setConsumer(new DelegatingProgressBarConsumer(LOGGER::info));
+        try (
+            OutputStream fileOS = Files.newOutputStream(dst.toPath());
+            InputStream is = url.openStream();
+            InputStream pbis = ProgressBar.wrap(is, pbb);
+            InputStream bis = new BufferedInputStream(pbis)) {
+          IOUtils.copyLarge(bis, fileOS);
+          return true;
+        }
+      } catch (IOException e) {
+        LOGGER.info("Unable to download from {}", url, e);
+        ++retry;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @param livyVersion
+   * @param targetLivyHomeFolder
+   * @return livyHome
+   */
+  public static String downloadLivy(String livyVersion, File targetLivyHomeFolder) {
+    File livyDownloadFolder = new File(downloadFolder, "livy");
+    livyDownloadFolder.mkdir();
+    if (targetLivyHomeFolder.exists()) {
+      LOGGER.info("Skip to download Livy {} as it is already downloaded.", livyVersion);
+      return targetLivyHomeFolder.getAbsolutePath();
+    }
+    File livyZip = new File(livyDownloadFolder, "livy-" + livyVersion + ".zip");
+    try {
+      URL mirrorURL = new URL(
+          IOUtils.toString(new URL(MIRROR_URL), StandardCharsets.UTF_8) + "incubator/livy/"
+              + livyVersion
+              + "/apache-livy-" + livyVersion + "-bin.zip");
+      URL archiveURL = new URL("https://archive.apache.org/dist/incubator/livy/" + livyVersion
+          + "/apache-livy-" + livyVersion + "-bin.zip");
+      LOGGER.info("Download Livy {}", livyVersion);
+      download(new DownloadRequest(mirrorURL, archiveURL), livyZip);
+      LOGGER.info("Unzip Livy {} to {}", livyVersion, targetLivyHomeFolder);
+      ProgressBarBuilder pbb = new ProgressBarBuilder()
+          .setTaskName("Unarchiv Livy")
+          .setUnit("MiB", 1048576) // setting the progress bar to use MiB as the unit
+          .setStyle(ProgressBarStyle.ASCII)
+          .setUpdateIntervalMillis(1000)
+          .setConsumer(new DelegatingProgressBarConsumer(LOGGER::info));
+      try (InputStream fis = Files.newInputStream(livyZip.toPath());
+          InputStream pbis = ProgressBar.wrap(fis, pbb);
+          InputStream bis = new BufferedInputStream(pbis);
+          ZipInputStream zis = new ZipInputStream(bis)) {
+        unzip(zis, targetLivyHomeFolder, 1);
+      }
+      LOGGER.info("Unzip Livy {} done", livyVersion);
+      // Create logs directory
+      File logs = new File(targetLivyHomeFolder, "logs");
+      logs.mkdir();
+    } catch (MalformedURLException e) {
+      LOGGER.error("invalid URL", e);
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to download livy", e);
+    }
+    return targetLivyHomeFolder.getAbsolutePath();
+  }
+
+  /**
+   * @param livyVersion
+   * @return return livyHome
+   * @throws IOException
+   */
+  public static String downloadLivy(String livyVersion) {
```

Comment:
livy has scala suffix since 0.8, see https://github.com/apache/zeppelin/pull/4678

as the current master uses 0.7, leave it also fine